063-topic-replication-factor-change.md

```diff
@@ -0,0 +1,221 @@
+This map is used to periodically check Cruise Control's tasks and update the KafkaTopic status.
+
+#### Configuration
+
+To request a replicas change, the Topic Operator needs to authenticate and send HTTP requests to the Cruise Control's REST API.
+The following environment variables are set automatically by the Cluster Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment.
+
+| Name                                   | Type    | Default | Description                                                   |
+|:---------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED         | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_NAMESPACE       | String  | ""      | Namespace where Cruise Control is running                     |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME        | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT            | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED     | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED    | Boolean | false   | Whether Basic authentication is enabled                       |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED    | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_SECRET_NAME     | String  | ""      | Secret name containing the truststore                         |
+| STRIMZI_CRUISE_CONTROL_API_SECRET_NAME | String  | ""      | Secret name containing the API credentials                    |
+
+In case of Cluster Operator deployment, a dedicated admin user is automatically created for the Topic Operator.
+
+In case of standalone deployment, the Cluster Operator expects the following secrets to be created by the user:
+
+- A secret with `cruise-control.p12` key containing the truststore in PKCS12 format, and `cruise-control.password` key containing the truststore password.
+- A secret with `cruise-control.apiToAdminUsername` key containing the API username with admin role, and `cruise-control.apiToAdminPassword` key containing the user password.
+
+#### Error handling
+
+A replicas change is indefinitely retried by the periodic reconciliation when Cruise Control's task queue is full, or there is an error.
+The user is always informed about the replicas change state through appropriate logs and KafkaTopic status updates.
+
+During the whole replicas change process, the KafkaTopic resource remains in a Ready state.
+Any KafkaTopic with replicas change has a status condition of type Ongoing and reason ReplicasChanged, which contains the CruiseControl `User-Task-ID` in the message.
```

Comment:
Seems weird to hide this in status condition. I think oyu should consider a separate subsection in the KafkaTopic CR status section. (For the Task ID => condition is fine to inform about issues and progress etc.)

Reply:
Ok, that's a good idea. I was just waiting for someone to raise it :)